ql/src/java/org/apache/hadoop/hive/ql/metadata/HiveStorageHandler.java

```diff
@@ -99,6 +99,12 @@ public abstract void configureInputJobProperties(TableDesc tableDesc,
     Map<String, String> jobProperties);
 
   /**
+   * This method is called to allw the StorageHandlers the chance to
+   * populate secret keys into the job's credentials.
+   */
+  public abstract void configureInputJobCredentials(TableDesc tableDesc, Map<String, String> secrets);
```

Comment:
We should call out the api change in the release notes section of the jira and check the backward incompatible flag